modules/nf-core/sourmash/index/main.nf

```diff
@@ -0,0 +1,34 @@
+process SOURMASH_INDEX {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::sourmash=4.6.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.6.1--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.6.1--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(signature)
+
+    output:
+    tuple val(meta), path("*.sbt.zip"), emit: signature_index
+    path "versions.yml"               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: "--dna -k 31"
```

Comment:
What's this do? You shouldn't set any arguments here IMO.

Reply:
It won't run without specifying either `--dna` or  `--protein` as well as a value for `-k` (even though the documentation says that 31 is default), therefore I had to specify something. But maybe I should specify them somewhere else? In the `tests/modules/nf-core/sourmash/index/test.yml`?

Reply:
The general consensus it that we can't assume anything in the module. Any mandatory stuff should be documented in `meta.yml`.

For the tests you can edit `ext.args` in `nextflow.config`

Reply:
To be sure I understand, I should not declare them in `main.nf` but instead document that they're required in `meta.yml` and specify them in `nextflow.config`?

Reply:
Yes on all accounts!